### ISSUE TITLE
Handling of invalid calculator results

### ISSUE THREAD

**Issue by pstorch** [cgeo/cgeo#6855]
There are some situations where the calculator results are wrong, especially when the inputs are invalid.

For example the minutes are out of range, then the leading zeros are not added:
![image](https://user-images.githubusercontent.com/5598124/33243988-da508cd8-d2ef-11e7-901c-fe7674df186c.png)

The calculated decimals result in more than expected digits, e.g. 4 digits for the decimal minutes instead of 3:
![image](https://user-images.githubusercontent.com/5598124/33244053-ddd3669a-d2f0-11e7-82a3-16096508b57d.png)

I don't think we have to solve this before the next release. It's just a successor of #6842 with some more ambiguous inputs.

**Comment by Matze2** [comment]
I think the first case is ok since the result is gray (=invalid).
The second case should also lead to a gray result.

I had a case where an 'x' was taken as multiplication operator sign: (Ax3-2)
Took some time to find the problem. But the result was also gray.
Same for negative numbers.

A different question is if a grayed out result font is enough indication for being invalid. Maybe an icon, e.g. 👍 or 👎, or different background colors would help here.

**Comment by S-Bartfast** [comment]
>For example the minutes are out of range, then the leading zeros are not added:

Yeah, went back and forth as to what to show when coordinates are not valid.  At the moment the original input string is shown with appropriate letter substitutions, but bracketed sections are not replaced and padded zeros are not inserted.  I'm happy to change what is shown for invalid results and indeed I'm keen to provide more precise feedback to the user but this is what I thought was best at the moment –though I'd be happy to change it if you think something else would be better.

>The calculated decimals result in more than expected digits, e.g. 4 digits for the decimal minutes instead of 3:

Well, this one is a parser issue as well.  If we want additional decimal digits to make a formulation invalid that's a change we should probably make in the GeoPoint parser rather than the calculator specifically.  While there is indeed an argument to be made as to additional digits being _more_ of a problem for the calculator I'd prefer not to make too many unique rules for the calculator.

Note when the coordinate's value is valid the result that is shown comes from the GeoPoint rather than the calculator itself

>A different question is if a grayed out result font is enough indication for being invalid.

The "Done" button (the 'Tick' mark in the top corner) also changes to a 'disk' icon to indicate the calculators state can be 'saved'.  While the appearance of the disk icon might not be enough to convey the fact that the coordinates are invalid the fact that the 'tick' button is gone and the only way to exit is now to 'Cancel' should be enough to let the user know they don't have valid coordinates.

**Comment by Lineflyer** [comment]
> The "Done" button (the 'Tick' mark in the top corner) also changes to a 'disk' icon to indicate the calculators state can be 'saved'. While the appearance of the disk icon might not be enough to convey the fact that the coordinates are invalid the fact that the 'tick' button is gone and the only way to exit is now to 'Cancel' should be enough to let the user know they don't have valid coordinates.

This might be superseeded given the discussion in #6903

**Comment by Lineflyer** [comment]
@eddiemuc 

Same here: Maybe you can take a look if this is still relevant while working in that area.

**Comment by eddiemuc** [comment]
If I understand the issue correctly, then the request is to mark calculation results as invalid if e.g. a DD MM.MMM format calculation results in more than 3 subdigits in Minutes. I have a different opinion about that: IMHO this should be considered a valid coordinate (like it already is in todays implementation)
